### PR TITLE
added `abi3-py312` feature

### LINKED ISSUE #3645: Please add an `abi3-py312` feature flag
- Without this feature flag I cannot use PyO3 with a custom built libpython3.12. 
 - This is a monkey-see monkey-do 5-10 min task similar to https://github.com/PyO3/pyo3/pull/2776/files
 
 I have this working locally, but I cannot post a PR, as my company is in transition and previously unapproved OSS contribution is on hold.

### PR DESCRIPTION
**Pull request by alex**
Fixes #3645

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -88,6 +88,7 @@ abi3-py38 = ["abi3-py39", "pyo3-build-config/abi3-py38", "pyo3-ffi/abi3-py38"]
 abi3-py39 = ["abi3-py310", "pyo3-build-config/abi3-py39", "pyo3-ffi/abi3-py39"]
 abi3-py310 = ["abi3-py311", "pyo3-build-config/abi3-py310", "pyo3-ffi/abi3-py310"]
 abi3-py311 = ["abi3", "pyo3-build-config/abi3-py311", "pyo3-ffi/abi3-py311"]
```
Comment (adamreichold) [inline Cargo.toml:90]: This should require `abi3-py312` instead of `abi3` then.

File: pyo3-ffi/Cargo.toml [hunk 2]
```diff
@@ -32,6 +32,7 @@ abi3-py38 = ["abi3-py39", "pyo3-build-config/abi3-py38"]
 abi3-py39 = ["abi3-py310", "pyo3-build-config/abi3-py39"]
 abi3-py310 = ["abi3-py311", "pyo3-build-config/abi3-py310"]
 abi3-py311 = ["abi3", "pyo3-build-config/abi3-py311"]
```
Comment (adamreichold) [inline pyo3-ffi/Cargo.toml:34]: This should require `abi3-py312` instead of `abi3`.

**Comment by alex** [comment]
Doh, good catch.

**Review by davidhewitt — APPROVED** [review]
LGTM, thanks!👍

**Pull request merged**